executor/runcexecutor/executor.go

```diff
@@ -220,10 +221,30 @@ func (w *runcExecutor) Exec(ctx context.Context, meta executor.Meta, root cache.
 		return err
 	}
 
+	done := make(chan struct{})
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				if err := w.runc.Kill(context.TODO(), id, int(syscall.SIGKILL), nil); err != nil {
```

Comment:
you should maybe use a context with Timeout when calling runc.Kill just to make sure that command does not hang

Reply:
Updated